crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1089,37 +1104,35 @@ where
                 cases,
                 range: _,
             }) => {
+                debug_assert!(self.current_match_case.is_none());
```

Comment:
Nit: Using `assert_eq` gives more useful assertion messages if `current_match_case` is `Some` for whatever reason
```suggestion
                debug_assert_eq!(self.current_match_case.is_none());
```